utilities/similarity/entrypoint.py

```diff
@@ -4,16 +4,113 @@
 import time
 import random
 import tempfile
+import subprocess
 
 from nmtwizard.utility import Utility
 from nmtwizard.logger import get_logger
 from nmtwizard.utility import resolve_environment_variables
 from nmtwizard.storage import StorageClient
 
 from similarity import main
+from build_vocab import main as build_vocab
+from build_data import main as build_data
+# import pyonmttok after tensorflow
+import pyonmttok
 
 logger = get_logger(__name__)
 
+tools_dir = os.getenv('TOOLS_DIR', '/root/tools')
+
+def setCUDA_VISIBLE_DEVICES(gpuid):
+    if gpuid == 0:
+        os.environ['CUDA_VISIBLE_DEVICES'] = "-1"
+    else:
+        if isinstance(gpuid, list):
+            os.environ['CUDA_VISIBLE_DEVICES'] = ",".join(str(i - 1) for i in gpuid)
+        else:
+            os.environ['CUDA_VISIBLE_DEVICES'] = str(gpuid - 1)
+        logger.debug(' - set CUDA_VISIBLE_DEVICES= %s' % os.environ['CUDA_VISIBLE_DEVICES'])
+
+def train_joint_tok_model(file_src, file_tgt):
+    filenames = [file_src, file_tgt]
```

Comment:
You can directly call `ingest_file` on `file_src` then on `file_tgt`. `SentencePieceLearner` keeps its own temporary file internally.